test/addons/callback-scope/binding.cc

```diff
@@ -0,0 +1,39 @@
+#include "node.h"
+#include "v8.h"
+
+#include <assert.h>
+#include <vector>
+
+namespace {
+
+void RunInCallbackScope(const v8::FunctionCallbackInfo<v8::Value>& args) {
+  v8::Isolate* isolate = args.GetIsolate();
+
+  assert(args.Length() == 4);
+  assert(args[0]->IsObject());
+  assert(args[1]->IsNumber());
+  assert(args[2]->IsNumber());
+  assert(args[3]->IsFunction());
+
+  node::async_context asyncContext = {
+    args[1].As<v8::Number>()->Value(),
+    args[2].As<v8::Number>()->Value()
```

Comment:
Well, yes, but that in turn is because `NumberValue()` actually performs a conversion to number if the argument is not already a number, possibly invoking userland code.

Since at this point we know that the arguments are `Number` instances, that’s redundant.